### PR TITLE
chore(text-field, combobox, number-field): refactor focus (VIV-1572)

### PR DESCRIPTION
**Pull request by rachelbt**

### CODE REVIEW

File: libs/components/src/lib/text-field/text-field.scss [hunk 1]
```diff
@@ -196,16 +192,26 @@ $low-ink-color: --_low-ink-color;
...
-	// only text-field has wrapper.
+		@include focus.focus;
+	}
+}
 
-	.wrapper:not(:focus-visible, :focus-within) > &,
-	.fieldset:not(:focus-visible, :focus-within) > & {
-		display: none;
+@supports not selector(:has(*)) {
+	.wrapper:focus-within::after {
```
Comment (RichardHelm) [inline libs/components/src/lib/text-field/text-field.scss:206]: If this code works fine, why add the one with `:has` as well? It isn't even any simpler

File: libs/components/src/lib/combobox/combobox.scss [hunk 2]
```diff
@@ -50,3 +51,28 @@
...
+@supports selector(:has(*)) {
+.fieldset:has(.control:focus-within)::after {
+	#{focus-variables.$focus-stroke-gap-color}: transparent;
+
+	@include focus.focus;
+}
+}
+
+@supports not selector(:has(*)) {
+.fieldset:focus-within::after {
```
Comment (RichardHelm) [inline libs/components/src/lib/combobox/combobox.scss:66]: We could consider adding wrapper element to combobox as well to simplify the CSS

File: libs/components/src/shared/patterns/focus.ts
Comment (RichardHelm) [inline libs/components/src/shared/patterns/focus.ts:1]: 🎉

**Review by RichardHelm — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/text-field/text-field.scss (hunk 1 above)
Comment (rachelbt) [reply libs/components/src/lib/text-field/text-field.scss:206]: bacause I do not like the face that we add an addition pseudo element. Its the right way to use has. I guess in 1/2 a year or so we can remove the fallback

File: libs/components/src/lib/combobox/combobox.scss (hunk 2 above)
Comment (rachelbt) [reply libs/components/src/lib/combobox/combobox.scss:66]: Its a good question wether we should add it.
We don't need an addition HTML wrapper, but on the other hand it helps consistency and cleaner css.
Either way - I wouldn't want to do thin in this PR.

File: libs/components/src/lib/text-field/text-field.scss (hunk 1 above)
Comment (RichardHelm) [reply libs/components/src/lib/text-field/text-field.scss:206]: But the `:has` code is also using a pseudo element 🤔

Comment (rachelbt) [reply libs/components/src/lib/text-field/text-field.scss:206]: that we already have :) Not creating a new one for the focus

**Review by RichardHelm — APPROVED** [review]

**Pull request merged**